components/ChallengeMaterial.tsx

```diff
@@ -304,8 +304,10 @@ const ChallengeMaterial: React.FC<ChallengeMaterialProps> = ({
     },
     {}
   )
+  //chain sort-map causes errors in tests without a copy
+  const challengesCopy = [...challenges]
   //create a new Challenges array with user submission data integrated and sorted
-  const challengesWithSubmissionData: ChallengeSubmissionData[] = challenges
+  const challengesWithSubmissionData: ChallengeSubmissionData[] = challengesCopy
```

Comment:
I am surprised why this didn't caused any problems before. I created copy of challenges to use in map-sort, otherwise it was causing type errors.